### DISCUSSION TITLE
ValueError: y_pred and y should have same shapes

### DISCUSSION THREAD

**Discussion by oscarramirezs** [Project-MONAI/tutorials#567] [Q&A] [answered]
Hello

I'm using the  "spleen_segmentation_3d.ipynb"on my dataset.

My Shoulder dataset included images and labels (0:background, 1:Humerus , 2: Scapula)

I changed model as
model = UNet(
dimensions=3,
in_channels=1,
out_channels=3,
channels=(16, 32, 64, 128, 256),
strides=(2, 2, 2, 2),
num_res_units=2,
norm=Norm.BATCH,
).to(device)
and 
post_pred = Compose([EnsureType(), AsDiscrete(argmax=True, to_onehot=3)])
post_label = Compose([EnsureType(), AsDiscrete(to_onehot=3)])

When I run code, the training occurs well but,  it occured a bug in valiadation process:   ValueError: y_pred and y should have same shapes
_______________
C:\ProgramData\Anaconda3\lib\site-packages\monai\metrics\meandice.py in compute_meandice(y_pred, y, include_background)
    121 
    122     if y.shape != y_pred.shape:
--> 123         raise ValueError("y_pred and y should have same shapes.")
    124 
    125     # reducing only spatial dimensions (not batch nor channels)

ValueError: y_pred and y should have same shapes
______________

i dont know why the 

val_labels.shape is =  torch.Size([3, 1, 130, 130, 97])
while the 
val_outputs.shape is = torch.Size([1, 3, 130, 130, 97])

can someone help me?  I'll appreciate it very much

thanks

**Comment by Yangtting** [comment, marked as answer]
Please update to run with MONAI v0.8.0 again. I hope I can help you.

**Reply by HripsimeS** [reply]
@oscarramirezs Hello. Did you find a solution for your issue? I am having the same problem during testing process.

**Reply by oscarramirezs** [reply]
Hello 

I update monai to version 8 and it works fine.
